core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java

```diff
@@ -144,4 +157,21 @@ private void addClassTypeHierarchy(ReflectiveHierarchyBuildItem reflectiveHierar
     private boolean skipClass(DotName name, Predicate<DotName> ignorePredicate, Set<DotName> processedReflectiveHierarchies) {
         return ignorePredicate.test(name) || processedReflectiveHierarchies.contains(name);
     }
+
+    private boolean doFinalFieldsNeedToBeWritable(ClassInfo classInfo) {
+        if (classInfo == null) {
+            return false;
+        }
+        if (finalFieldsWritablePredicates.isEmpty()) {
+            return false;
+        }
+        if (finalFieldsWritablePredicate == null) {
+            // create a predicate that returns true if any of the predicates says that final fields need to be writable
+            finalFieldsWritablePredicate = finalFieldsWritablePredicates
+                    .stream()
+                    .map(ReflectiveClassFinalFieldsWritablePredicateBuildItem::getPredicate)
+                    .reduce(c -> false, Predicate::or);
+        }
```

Comment:
I don't think I would have done it this way. Can't we build the predicate once where we need it and pass it as a parameter?

Reply:
You mean instead of stuffing the result in the `private static` field?

Reply:
Yes.

Reply:
Good idea, fixed